### ISSUE TITLE
not working

### ISSUE THREAD

**Issue by luciofncnd** [waaah/pitchfinder_v2#1]
Hi, I compiled to a Pixel 2 without problems, and seems to work ok, but doesn't detect anything.

**Comment by ixugo** [comment]
I tried it , you are right , not working

**Comment by christianparpart** [comment]
To bad, i just wanted to give it a try m why updates? @waaah are you still maintaining this?

**Comment by tuntorius** [comment]
It works for me. Take a look at the example https://github.com/waaah/pitchfinder_v2/blob/master/example/lib/main.dart
you probably missed that you need to listen to onRecorderStateChanged stream. Don't use pitch variable in PitchDetector class. It's always null.

**Comment by maestrocoder** [comment]
my experience is that ios works - the example shows the pitch in Hz
android does NOT work, instead where android should return a double it is returning _Float32ArrayView an array of floats so this is just a theory but based on history 0.0.3 when author added PCM ios support maybe some regression bug.

to test this hypothesis...

@tuntorius are you on ios?
@ixugo @luciofncnd are you on android?

**Comment by maestrocoder** [comment]
found this - going to see if it fixes and if so will submit MR

![2022-03-04_21-09-52](https://user-images.githubusercontent.com/2679210/156869064-f6f03014-4d2d-4fa9-b243-49ccc26eb613.png)

on android this would explain why onRecorderStateChanged stream is showing the null because it has an error and the pitchResult does not need to be samples, it should return the pitchResult

just a theory, hoping to be wrong

**Comment by maestrocoder** [comment]
yes , just change this one line of code in (PitchdetectorPlugin.java):

![2022-03-04_21-20-23](https://user-images.githubusercontent.com/2679210/156869612-c56da83a-b6f7-4bc6-8eb8-f149e471524d.png)

then you get pitch in Hz on android:

![2022-03-04_21-20-48](https://user-images.githubusercontent.com/2679210/156869629-ef6758ea-8d1d-4b75-826e-d18516e69374.png)

**Comment by maestrocoder** [comment]
MR#2 fix for this https://github.com/waaah/pitchfinder_v2/pull/2

or see the one liner:
channel.invokeMethod("getPitch", pitchResult);
